pkg/xds/translate.go

```diff
@@ -137,26 +309,22 @@ func TranslateRoute(in *routingv2.HTTPRoute,
 			action.RequestMirrorPolicy = &route.RouteAction_RequestMirrorPolicy{Cluster: name(in.Mirror)}
 		}
 
-		if len(in.Route) == 0 { // build default cluster
-			action.ClusterSpecifier = &route.RouteAction_Cluster{Cluster: defaultCluster}
-		} else {
-			weighted := make([]*route.WeightedCluster_ClusterWeight, len(in.Route))
-			for _, dst := range in.Route {
-				weighted = append(weighted, &route.WeightedCluster_ClusterWeight{
-					Name:   name(dst.Destination),
-					Weight: &types.UInt32Value{Value: uint32(dst.Weight)},
-				})
-			}
+		weighted := make([]*route.WeightedCluster_ClusterWeight, len(in.Route))
+		for _, dst := range in.Route {
+			weighted = append(weighted, &route.WeightedCluster_ClusterWeight{
+				Name:   name(dst.Destination),
+				Weight: &types.UInt32Value{Value: uint32(dst.Weight)},
```

Comment:
This is where the metadata struct should go in.

Reply:
I understand how it _could be done_ and actually agree with you that's it's better. It's just not the goal.